### DISCUSSION TITLE
How to get all objects that match certain value in non-id field

### DISCUSSION THREAD

**Discussion by kruppy** [encode/django-rest-framework#8164] [Question & Answer]
Hello,

when trying to query objects on their foreignkey field, I dont manage to get any details. The company ID is and uuid (uuid4) field.

I have a Model called "contacts":
```python
class Contact(models.Model):
    firstname = models.CharField(max_length=35)
    lastname = models.CharField(max_length=35)
    company = models.ForeignKey(Company, on_delete=models.CASCADE)
```

I want to get all Contacts, that work for the same company. Therefore I have created a ListAPIView whitin views.py
Views.py
```python
class ContactViewSet(viewsets.ModelViewSet):
    queryset = Contact.objects.all()
    serializer_class = ContactSerializer


class CompanyContactsListView(generics.ListAPIView):
    serializer_class = ContactSerializer

    def get_queryset(self):
        company = self.kwargs['company']
        return Contact.objects.filter(company=company)
```

And to get a URL I added the path in urls.py
```python 
urlpatterns = [
    path('', include(router.urls)),
    path('contacts/<uuid:company>/', CompanyContactsListView.as_view(), name='contacts')
]
```

Problem is, that when i try to go for that path and enter an UUID of an company that exists and has related contacts, I get the following error
```python
HTTP 404 Not Found
Allow: GET, PUT, PATCH, DELETE, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "detail": "Not found."
}
```

Is it possible, that my URL is wrong and therefore cant query the ListAPIView?

Thanks in advance!

**Comment by phuongtai** [comment]
The field `company` in your Contact not a UUID field, so that's why the url will be not found.

**Reply by kruppy** [reply]
Thank you for your answer, phuongtai!
I am a little bit confused, because when receiving the Contacts, it seems that in company there is a uuid stored:
```json
[
    {
        "id": 1,
        "firstname": "Joshuah",
        "lastname": "Bankhurst",
        "company": "e871c47b-9b91-4cf9-94a6-e8135510c11d"
    },
    {
        "id": 2,
        "firstname": "Clayborn",
        "lastname": "Sylett",
        "company": "e871c47b-9b91-4cf9-94a6-e8135510c11d"
    }
]
```

So is this not a uuid?

**Comment by kruppy** [comment]
hi guys, I added the code in a repository. Maybe it helps to recreate my problem...
https://github.com/kruppy/endpoint_test
